File: 해시/지현/전화번호_목록.py
def solution(phone_book):
    #문자열 기준으로 정렬 + 다음거랑 비교
    phone_book.sort()

    for i in range(len(phone_book)-1):
        length= len(phone_book[i])
        if len(phone_book[i])<=len(phone_book[i+1]):       
            if phone_book[i+1][0:length]==phone_book[i]:
                return False
    return True

File: 해시/[영우] 해시/전화번호_목록.js
function solution(phone_book) {
  // 전화번호를 순서대로 정렬한다
  // js 의 sort 는 유니코드 순서를 따르기 때문에, 앞글자부터 비교가 가능하다
  phone_book.sort();

  for (let i = 0; i < phone_book.length; i++) {
    const phone1 = String(phone_book[i]);
    const phone2 = String(phone_book[i + 1]);

    // 뒤의 값이 앞의 값으로 시작하면 false 를 반환한다
    if (phone2.startsWith(phone1)) {
      return false;
    }
  }

  // 전부 통과하면 true 를 반환한다
  return true;
}
